### ISSUE TITLE
Compiled code fails server-side

### ISSUE THREAD

**Issue by damusnet** [franciscop/use-animation-frame#1]
Hi @franciscop!

This is a nice and handy hook, thanks for publishing it!

I think I found a minor bug: it crashes when imported in a Next.js project like this:
![image](https://user-images.githubusercontent.com/433409/93117725-82429280-f67c-11ea-827f-4b6ba9892c7f.png)

I believe this is due to how the code is compiled, since there is no reference to `window` in the source code, but the compiled and minified code looks like this:

```js
!(function (e, t) {
  "object" == typeof exports && "object" == typeof module
    ? (module.exports = t(require("react")))
    : "function" == typeof define && define.amd
    ? define("use-animation-frame", ["React"], t)
    : "object" == typeof exports
    ? (exports["use-animation-frame"] = t(require("react")))
    : (e["use-animation-frame"] = t(e.React));
})(window, function (e) {
  return (function (e) {
    var t = {};
    ...
```

Thankfully, there is an easy work around due to the fact that you included the source in the npm package:
```diff
- import useAnimationFrame from 'use-animation-frame'
+ import useAnimationFrame from 'use-animation-frame/index'
```

I'm not super up to date on library compilation techniques, and I don't know if it's best to update the `package.json` to point to the uncompiled code in a special field, or to compile differently to avoid the `window`, or, I think I have also seen other libraries treat the `useEffect` case as an exception so that it becomes a true noop on the server.

Hope this helps!

**EDIT:** unfortunately, while my quick fix worked in `dev`, it fails in `prod`, as our build system does not compile `node_modules` out of the box, and it seems like more trouble than is worth to change that, instead of copy/pasting this lib.

**Comment by EricWVGG** [comment]
I crashed into this same problem, it prohibits this lib from being used with Gatsby and presumably Next.js or any other SSR setup. 

Everything appears to work fine if the following lines are inserted into the beginning of the main function block:
```
  if( typeof performance === 'undefined' || typeof window === 'undefined' ) {
    return
  }
```

otherwise works perfectly! Thank you for publishing this work.

**Comment by franciscop** [comment]
Thanks for opening an Issue with a potential bug fix, please feel free to send a PR. Please include some way to reproduce it online (before/after?), since I don't use Gatsby/Next.js.

**Comment by EricWVGG** [comment]
I've put together a PR and a test that reproduces the issue, but I'm too dumb to figure out the minification settings you used to make the min.js file. Could you tell me which tool and parameters you used?

**Comment by franciscop** [comment]
Thanks! The settings are in [`webpack.config.json`](https://github.com/franciscop/use-animation-frame/blob/master/webpack.config.js) and the command to run that is in [`package.json`](https://github.com/franciscop/use-animation-frame/blob/master/package.json): `"build": "webpack -c"`

**Comment by franciscop** [comment]
BTW, now that I think about it, it might not be needed to transpile this library (if we use _only_ ES6+ code and no JSX): https://stackoverflow.com/a/59655473/938236. Let's compile it for now to fix this issue, and I'll run some experiments and consider whether we should not transpile it in the future!

**Comment by hazem3500** [comment]
Is this still planned to be merged?

**Comment by franciscop** [comment]
Thanks for the reminder! Published as `use-animation-frame@0.1.1`, since I've never used Next.js please let me know if there's any issue remaining! (reopen this or open a new issue)
